log4j-core/src/test/java/org/apache/logging/log4j/junit/DisableOnWindows.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache license, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the license for the specific language governing permissions and
+ * limitations under the license.
+ */
+package org.apache.logging.log4j.junit;
+
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.RetentionPolicy;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+@Target({ ElementType.TYPE, ElementType.METHOD })
+@Retention(RetentionPolicy.RUNTIME)
+@ExtendWith(DisableOnWindowsCondition.class)
+public @interface DisableOnWindows {
```

Comment:
Should we make this more general? DisableOnOS? Then we'd have @DisableOnOS("windows") rather than @DisableOnWindows. Also, is this the correct place in the project structure to put this helper class? Or should it get moved in with other classes that help with testing? If so, where would that be?

Reply:
It looks like I should  have just used @DisabledOnOs(WINDOWS). I'll drop these new classes and switch it to use org.junit.jupiter.api.condition.DisabledOnOs

Reply:
I switched the JUnit @DisabledOnOs(WINDOWS), built, tested, and pushed to github.